test/CourseModule/course.e2e-spec.ts

```diff
@@ -0,0 +1,60 @@
+import * as request from 'supertest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication, ValidationPipe } from '@nestjs/common';
+import { AppModule } from '../../src/app.module';
+import { Connection, EntityManager, QueryRunner, Repository } from 'typeorm';
+import { ClientCredentials } from '../../src/SecurityModule/entity';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { ClientCredentialsEnum, GrantTypeEnum, RoleEnum } from '../../src/SecurityModule/enum';
+import { Constants } from '../../src/CommonsModule';
+
+const stringToBase64 = (string: string) => {
+  return Buffer.from(string).toString('base64');
+};
+
+describe('CourseController (e2e)', () => {
+  let app: INestApplication;
+  let moduleFixture: TestingModule;
+  let queryRunner: QueryRunner;
+  let authorization: string;
+  const CourseUrl: string = `/${Constants.API_PREFIX}/${Constants.API_VERSION_1}/${Constants.COURSE_ENDPOINT}`;
+
+  beforeAll(async () => {
+    moduleFixture = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe());
+    await app.init();
+
+    const dbConnection = moduleFixture.get(Connection);
+    const manager = moduleFixture.get(EntityManager);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    queryRunner = manager.queryRunner = dbConnection.createQueryRunner('master');
+
+    const clientCredentialRepository: Repository<ClientCredentials> = moduleFixture.get<Repository<ClientCredentials>>(getRepositoryToken(ClientCredentials));
+    const clientCredentials: ClientCredentials = new ClientCredentials();
+    clientCredentials.name = ClientCredentialsEnum.FRONT;
+    clientCredentials.secret = 'test2';
+    clientCredentials.role = RoleEnum.ADMIN;
+    await clientCredentialRepository.save(clientCredentials);
+    authorization = stringToBase64(`${clientCredentials.name}:${clientCredentials.secret}`);
+  });
+
```

Comment:
Falta os testes batendo nos endpoints pra saber se ta funcionando certinho

Reply:
Como eu so fiz os metodos Get, eu nao consigo dar um post pra verificar se o usuario foi criado conforme o test do user

`  it('should find user by id', async () => {
    return request(app.getHttpServer())
      .post('/oauth/token')
      .set('Authorization', `Basic ${authorization}`)
      .set('Content-Type', 'multipart/form-data')
      .field('grant_type', GrantTypeEnum.CLIENT_CREDENTIALS)
      .then((res) => {
        return request(app.getHttpServer())
          .post(userUrl)
          .set('Authorization', `Bearer ${res.body.accessToken}`)
          .send({
            email: 'my-user2@email.com',
            password: 'mypass',
            urlInstagram: 'instagram',
            urlFacebook: 'facebook',
            name: 'name',
            role: RoleEnum.ADMIN,
          } as NewUserDTO)
          .then((_res) => {
            return request(app.getHttpServer())
              .get(`${userUrl}/${_res.body.id}`)
              .set('Authorization', `Bearer ${res.body.accessToken}`)
              .expect((response) => {
                expect(response.body.email).toBe('my-user2@email.com');
              });
          });
      });
  });
`

Reply:
porque se eu so fizer assim vai dar erro correto?


```

  it('should find course by id', async () => {
    return request(app.getHttpServer())
      .post('/oauth/token')
      .set('Authorization', `Basic ${authorization}`)
      .set('Content-Type', 'multipart/form-data')
      .field('grant_type', GrantTypeEnum.CLIENT_CREDENTIALS)
      .then((res) => {
        return request(app.getHttpServer())
          .get(`${courseUrl}/`)
          .set('Authorization', `Bearer ${res.body.accessToken}`)
          .expect((response) => {
            expect(response.body.description).toBe('teste');
          });
      });
  });
```

Reply:
> Como eu so fiz os metodos Get, eu nao consigo dar um post pra verificar se o usuario foi criado conforme o test do user
> 
> ```
>  it('should find user by id', async () => {
>     return request(app.getHttpServer())
>       .post('/oauth/token')
>       .set('Authorization', `Basic ${authorization}`)
>       .set('Content-Type', 'multipart/form-data')
>       .field('grant_type', GrantTypeEnum.CLIENT_CREDENTIALS)
>       .then((res) => {
>         return request(app.getHttpServer())
>           .post(userUrl)
>           .set('Authorization', `Bearer ${res.body.accessToken}`)
>           .send({
>             email: 'my-user2@email.com',
>             password: 'mypass',
>             urlInstagram: 'instagram',
>             urlFacebook: 'facebook',
>             name: 'name',
>             role: RoleEnum.ADMIN,
>           } as NewUserDTO)
>           .then((_res) => {
>             return request(app.getHttpServer())
>               .get(`${userUrl}/${_res.body.id}`)
>               .set('Authorization', `Bearer ${res.body.accessToken}`)
>               .expect((response) => {
>                 expect(response.body.email).toBe('my-user2@email.com');
>               });
>           });
>       });
>   });
> ```

No beforeAll ta adicionando um clientCredentials na mão usando repository, faz do mesmo jeito pra adicionar um curso

Reply:
> porque se eu so fizer assim vai dar erro correto?
> 
> ```
> 
>   it('should find course by id', async () => {
>     return request(app.getHttpServer())
>       .post('/oauth/token')
>       .set('Authorization', `Basic ${authorization}`)
>       .set('Content-Type', 'multipart/form-data')
>       .field('grant_type', GrantTypeEnum.CLIENT_CREDENTIALS)
>       .then((res) => {
>         return request(app.getHttpServer())
>           .get(`${courseUrl}/`)
>           .set('Authorization', `Bearer ${res.body.accessToken}`)
>           .expect((response) => {
>             expect(response.body.description).toBe('teste');
>           });
>       });
>   });
> ```

Assim vai dar erro por dois motivos:
- Response.body vai ser um array, pra pegar alguma descrição cê tem que fazer response.body[0].description
- Como vai vir um array vazio, se você fizer response.body[0] vai dar undefined, se fizer response.body[0].description vai dar erro de "cannot find property description of undefined"